File: app/src/main/java/org/anvei/novelreader/widget/readview/loader/AbsBookLoader.java
package org.anvei.novelreader.widget.readview.loader;

public abstract class AbsBookLoader implements BookLoader {

    private String link;

    private String sourceName;          // 小说加载器书源名称
    private final int uid;              // 小说加载器唯一UID标识

    public AbsBookLoader(String sourceName, int uid) {
        this.sourceName = sourceName;
        this.uid = uid;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }

    public String getSourceName() {
        return sourceName;
    }

    public void setSourceName(String sourceName) {
        this.sourceName = sourceName;
    }

    public int getUid() {
        return uid;
    }

}

File: app/src/main/java/org/anvei/novelreader/loader/AbsBookLoader.kt
package org.anvei.novelreader.loader

import org.anvei.novelreader.loader.bean.SearchResultItem
import org.klee.readview.loader.BookLoader

abstract class AbsBookLoader(
    val uid: Int,           // 加载器唯一UID标识
    var name: String        // 加载器名称
) : BookLoader{

    var link: String? = null
    var o: Any? = null

    /**
     * 小说加载器的搜索功能
     * @param keyword
     * @return 搜索结果
     */
    abstract fun search(keyword: String): List<SearchResultItem>

    fun initToc(link: String) {
        this.link = link
        initToc()
    }

}